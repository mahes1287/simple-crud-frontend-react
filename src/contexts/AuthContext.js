import { createContext, useState } from "react";

const AuthContext = createContext();

export function AuthProvider({ children }) {
  return <AuthContext.Provider value={user}>{children}</AuthContext.Provider>;
}

export default AuthContext;
