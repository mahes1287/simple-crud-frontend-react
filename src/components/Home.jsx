import React from "react";
import { useAuth } from "../contexts/AuthContext";

export default function Home() {
  const { user } = useAuth();
  // console.log({ user });
  if (user) {
    return (
      <div
        data-testid="logged-in"
        className="container mx-auto text-center p-6"
      >
        <h1>
          Hello{" "}
          <span className="text-orange-500 text-xl">
            {localStorage.getItem("displayName")}
          </span>{" "}
          welcome to our app
        </h1>
        <div>Now you have a super power to add things !</div>
      </div>
    );
  }

  return (
    <div data-testid="logged-out" className="container mx-auto text-center p-6">
      You are at sweet home!!!!!
    </div>
  );
}
