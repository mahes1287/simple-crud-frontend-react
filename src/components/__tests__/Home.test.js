import Home from "../Home.jsx";
import { AuthContext } from "../../contexts/AuthContext.js";
import renderer from "react-test-renderer";
const { render, screen } = require("@testing-library/react");
describe("<Home />", () => {
  test("this should render homepage for logged out user", () => {
    const auth = { user: null };
    render(
      <AuthContext.Provider value={auth}>
        <div className="App">
          <Home />
        </div>
      </AuthContext.Provider>
    );

    const loggedOutElement = screen.getByTestId("logged-out");
    expect(screen.queryByTestId("logged-in")).not.toBeInTheDocument();
    expect(loggedOutElement).toBeInTheDocument();
    expect(loggedOutElement).toHaveTextContent("You are at sweet home!!!!!");
  });

  test("snapshot for logged out user", () => {
    const auth = { user: null };
    const tree = renderer
      .create(
        <AuthContext.Provider value={auth}>
          <div className="App">
            <Home />
          </div>
        </AuthContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("this should render homepage with logged in user", () => {
    const auth = {
      user: { uid: "jfjdffjvnhdf", displayName: "Maheshwaran Velusamy" },
    };
    localStorage.setItem("displayName", auth.user.displayName);
    render(
      <AuthContext.Provider value={auth}>
        <div className="App">
          <Home />
        </div>
      </AuthContext.Provider>
    );

    const loggedInElement = screen.getByTestId("logged-in");
    expect(loggedInElement).toBeInTheDocument();
    expect(screen.queryByTestId("logged-out")).not.toBeInTheDocument();
    expect(loggedInElement).toHaveTextContent(
      "Hello Maheshwaran Velusamy welcome to our app"
    );
  });
  test("snapshot for logged in user", () => {
    const auth = {
      user: { uid: "jfjdffjvnhdf", displayName: "Maheshwaran Velusamy" },
    };
    localStorage.setItem("displayName", auth.user.displayName);
    const tree = renderer
      .create(
        <AuthContext.Provider value={auth}>
          <div className="App">
            <Home />
          </div>
        </AuthContext.Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
