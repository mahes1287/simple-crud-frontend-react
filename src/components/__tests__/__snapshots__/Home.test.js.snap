// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> snapshot for logged in user 1`] = `
<div
  className="App"
>
  <div
    className="container mx-auto text-center p-6"
    data-testid="logged-in"
  >
    <h1>
      Hello
       
      <span
        className="text-orange-500 text-xl"
      >
        Maheshwaran Velusamy
      </span>
       
      welcome to our app
    </h1>
    <div>
      Now you have a super power to add things !
    </div>
  </div>
</div>
`;

exports[`<Home /> snapshot for logged out user 1`] = `
<div
  className="App"
>
  <div
    className="container mx-auto text-center p-10 space-y-3"
  >
    <span
      style={
        Object {
          "display": "inherit",
        }
      }
    >
      <span
        style={
          Object {
            "animation": "react-spinners-SyncLoader-sync 0.6s 0.07s infinite ease-in-out",
            "animationFillMode": "both",
            "backgroundColor": "rgba(54, 55, 214, 1)",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": "15px",
            "margin": "2px",
            "width": "15px",
          }
        }
      />
      <span
        style={
          Object {
            "animation": "react-spinners-SyncLoader-sync 0.6s 0.14s infinite ease-in-out",
            "animationFillMode": "both",
            "backgroundColor": "rgba(54, 55, 214, 1)",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": "15px",
            "margin": "2px",
            "width": "15px",
          }
        }
      />
      <span
        style={
          Object {
            "animation": "react-spinners-SyncLoader-sync 0.6s 0.21000000000000002s infinite ease-in-out",
            "animationFillMode": "both",
            "backgroundColor": "rgba(54, 55, 214, 1)",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": "15px",
            "margin": "2px",
            "width": "15px",
          }
        }
      />
    </span>
    <h1>
      Loading...
    </h1>
  </div>
</div>
`;
